### PR TITLE
Import `ascon-hash`

### PR DESCRIPTION
**Pull request by sebastinas**
The final step of https://github.com/sebastinas/ascon-aead/issues/5.

### CODE REVIEW

File: ascon-hash/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Changelog
+
+## 0.1.1 (2023-03-17)
+
+* Use `aead` instead of `aead-core`.
+* Bump MSRV to 1.60.
```
Comment (tarcieri) [inline ascon-hash/CHANGELOG.md:6]:
```suggestion
* Bump MSRV to 1.56.
```

Comment (tarcieri) [reply ascon-hash/CHANGELOG.md:6]: Oh wait, you already released 0.1.1

File: ascon-hash/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+    fn update_blocks(&mut self, blocks: &[Block<Self>]) {
+        for block in blocks {
+            self.state.absorb_block(block.as_ref());
+        }
+    }
+}
+
+/// Reader for XOF output
+#[derive(Clone, Debug)]
+pub struct AsconXOFReaderCore {
```
Comment (tarcieri) [inline ascon-hash/src/lib.rs]: To follow the UpperCamelCase naming convention:

```suggestion
pub struct AsconXofReaderCore {
```

This also better matches the [`digest::XofReader`](https://docs.rs/digest/latest/digest/trait.XofReader.html) trait

File: ascon-hash/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+
+impl AlgorithmName for AsconACore {
+    fn write_alg_name(f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.write_str("AsconAHash")
+    }
+}
+
+/// Ascon XOF
+#[derive(Clone, Debug, Default)]
+pub struct AsconXOFCore {
```
Comment (tarcieri) [inline ascon-hash/src/lib.rs]:
```suggestion
pub struct AsconXofCore {
```

File: ascon-hash/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+/// use ascon_hash::{AsconXOF, ExtendableOutput, Update, XofReader};
+///
+/// let mut xof = AsconXOF::default();
+/// xof.update(b"some bytes");
+/// let mut reader = xof.finalize_xof();
+/// let mut dst = [0u8; 5];
+/// reader.read(&mut dst);
+/// assert_eq!(&dst, b"\xc2\x19\x72\xfd\xe9");
+/// ```
+pub type AsconXOF = CoreWrapper<AsconXOFCore>;
```
Comment (tarcieri) [inline ascon-hash/src/lib.rs]:
```suggestion
pub type AsconXof = CoreWrapper<AsconXofCore>;
```

File: ascon-hash/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+/// let mut xof = AsconAXOF::default();
+/// xof.update(b"some bytes");
+/// let mut reader = xof.finalize_xof();
+/// let mut dst = [0u8; 5];
+/// reader.read(&mut dst);
+/// assert_eq!(&dst, b"\xb8\xd6\xbd\xf0\xa7");
+/// ```
+pub type AsconAXOF = CoreWrapper<AsconAXOFCore>;
+/// Reader for AsconAXOF output
+pub type AsconXOFReader = XofReaderCoreWrapper<AsconAXOFReaderCore>;
```
Comment (tarcieri) [inline ascon-hash/src/lib.rs]:
```suggestion
pub type AsconXofReader = XofReaderCoreWrapper<AsconAXofReaderCore>;
```

**Review by tarcieri — APPROVED** [review]

**Comment by tarcieri** [comment]
@sebastinas I added you to our hashes team. Can you add `github:rustcrypto:hashes` as a crate owner for `ascon-hash`?

**Comment by sebastinas** [comment]
```
$ cargo owner -a github:rustcrypto:hashes
    Updating crates.io index
       Owner team github:rustcrypto:hashes has been added as an owner of crate ascon-hash
```

**Pull request merged**